pkg/daemon/update.go

```diff
@@ -2155,23 +2155,26 @@ func (dn *Daemon) reboot(rationale string) error {
 	}
 	dn.logSystem("initiating reboot: %s", rationale)
 
-	rebootCmd := rebootCommand(rationale)
-
 	// reboot, executed async via systemd-run so that the reboot command is executed
 	// in the context of the host asynchronously from us
 	// We're not returning the error from the reboot command as it can be terminated by
 	// the system itself with signal: terminated. We can't catch the subprocess termination signal
 	// either, we just have one for the MCD itself.
-	if err := rebootCmd.Run(); err != nil {
-		dn.logSystem("failed to run reboot: %v", err)
+	rebootCmd := rebootCommand(rationale)
+	for {
```

Comment:
I agree here, the for loop seems a bit dangerous, since the MCD will no longer be able to reconcile any changes or react to updates that potentially clear this problem (although that is unlikely so it's not the end of the world).

Since we only sleep 5 minutes now, what happens if you remove this for loop entirely and just keep the contents inside? (or enqueue a node sync in 5 minutes instead of sleeping here)? I would think the MCD would try to update again and fail on this error, and so long as we do that within the 15 minutes the alarm would potentially still stay?

Reply:
Thanks for the feedback Jerry! Here are the changes I made

-removed the loop, left the 5 minute timeout in
-Added an immediate return if the command fails,  it didn't make sense for us to wait in this scenario. this still increments the counter(at a much faster pace), so it should still set the alert
-The other path is still a 5 minute timeout via a sleep
-Added/Changed some additional logging explaining both paths

the other thing to note is that once a reboot recovery happens(by this, I mean we have a fail, and then the reboot somehow works, not sure if this is even possible/realistic) - the alert will take 15 minutes to go away